node/src/components/contract_runtime/operations.rs

```diff
@@ -31,8 +31,10 @@ use casper_execution_engine::{
 };
 
 /// Executes a finalized block.
+#[allow(clippy::too_many_arguments)]
 pub fn execute_finalized_block(
     engine_state: &EngineState<LmdbGlobalState>,
+    system_contract_registry: &SystemContractRegistry,
```

Comment:
I don't think we should be passing `system_contract_registry` here.  In the `feat-fast-sync` branch, we only call `contract_runtime.set_initial_state` in the participating reactor, but I believe `execute_finalized_block` can be called from the chain synchronizer in the joiner reactor.  In that case, it would have an empty `SystemContractRegistry`.

We should probably change `get_era_validators` to take an optional `SystemContractRegistry`, and if it's `None` try to retrieve from the tracking copy.

Reply:
> In the feat-fast-sync branch, we only call `contract_runtime.set_initial_state` in the participating reactor, but I believe` execute_finalized_block` can be called from the chain synchronizer in the joiner reactor. In that case, it would have an empty `SystemContractRegistry`.

This PR is made to `dev`, not `feat-fast-sync` though.